block/v2/header.go

```diff
@@ -0,0 +1,417 @@
+package v2
+
+import (
+	"io"
+	"math/big"
+	"unsafe"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/rs/zerolog"
+
+	blockif "github.com/harmony-one/harmony/block/interface"
+	"github.com/harmony-one/harmony/crypto/hash"
+	"github.com/harmony-one/harmony/shard"
+)
+
+// Header is the V2 block header.
+type Header struct {
```

Comment:
can we use git cp (instead of cp then git add, which seems to have been used here) so that the change history log is maintained and we can see the changes made between v2/header.go vs. v1/header.go?

Reply:
Actually:

```
$ git cp
git: 'cp' is not a git command. See 'git --help'.

The most similar command is
        fap
$ 
```

Unlike Subversion and other VCS, Git does not track renames/moves/copies as first-class metadata.  The `git mv` command is simply a convenience shortcut for a `mv`–`git rm`–`git add` sequence, and the “renames” shown in `git log`, `git show`, and `git status` are actually detected by heuristics, which can be turned off:

```
$ git mv CONTRIBUTING.md CONTRIBUTING.mdx
$ git status
On branch master
Your branch is up to date with 'harmony-one/master'.

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        renamed:    CONTRIBUTING.md -> CONTRIBUTING.mdx

$ git status --no-renames # turn off rename detection
On branch master
Your branch is up to date with 'harmony-one/master'.

Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        deleted:    CONTRIBUTING.md
        new file:   CONTRIBUTING.mdx

$ 
```